lighthouse-core/scripts/i18n/collect-strings.js

```diff
@@ -44,52 +45,51 @@ function computeDescription(ast, property, startRange) {
 
 /**
  * @param {string} dir
- * @param {Record<string, ICUMessageDefn>} strings
+ * @return {Record<string, ICUMessageDefn>}
  */
-function collectAllStringsInDir(dir, strings = {}) {
-  for (const name of fs.readdirSync(dir)) {
-    const fullPath = path.join(dir, name);
-    const relativePath = path.relative(LH_ROOT, fullPath);
-    if (ignoredPathComponents.some(p => fullPath.includes(p))) continue;
-
-    if (fs.statSync(fullPath).isDirectory()) {
-      collectAllStringsInDir(fullPath, strings);
-    } else {
-      if (name.endsWith('.js')) {
-        if (!process.env.CI) console.log('Collecting from', relativePath);
-        const content = fs.readFileSync(fullPath, 'utf8');
-        const exportVars = require(fullPath);
-        const regexMatches = !!UISTRINGS_REGEX.test(content);
-        const exportsUIStrings = !!exportVars.UIStrings;
-        if (!regexMatches && !exportsUIStrings) continue;
-
-        if (regexMatches && !exportsUIStrings) {
-          throw new Error('UIStrings defined but not exported');
-        }
-
-        if (exportsUIStrings && !regexMatches) {
-          throw new Error('UIStrings exported but no definition found');
-        }
-
-        // @ts-ignore regex just matched
-        const justUIStrings = 'const ' + content.match(UISTRINGS_REGEX)[0];
-        // just parse the UIStrings substring to avoid ES version issues, save time, etc
-        // @ts-ignore - esprima's type definition is supremely lacking
-        const ast = esprima.parse(justUIStrings, {comment: true, range: true});
-
-        for (const stmt of ast.body) {
-          if (stmt.type !== 'VariableDeclaration') continue;
-          if (stmt.declarations[0].id.name !== 'UIStrings') continue;
-
-          let lastPropertyEndIndex = 0;
-          for (const property of stmt.declarations[0].init.properties) {
-            const key = property.key.name;
-            const message = exportVars.UIStrings[key];
-            const description = computeDescription(ast, property, lastPropertyEndIndex);
-            strings[`${relativePath} | ${key}`] = {message, description};
-            lastPropertyEndIndex = property.range[1];
-          }
-        }
+function collectAllStringsInDir(dir) {
+  /** @type {Record<string, ICUMessageDefn>} */
+  const strings = {};
+
+  const files = glob.sync(path.relative(LH_ROOT, dir) + '/**/*.js', {
```

Comment:
comment about glob only accepting `/`? I'm surprised that `path.relative` works on windows for this since it's going to have slashes going the wrong way....

Reply:
Evidently, glob doesn't only accept `/`. and I guess it does allow for mixed - path separators. I wonder if any comment at all would be more confusing than no comment :P

Reply:
Are we 100% positive it's working as expected...

https://github.com/isaacs/node-glob#windows is so convincing that only `/` works! 😆

Reply:
is `node-glob` === `glob`?

Reply:
anyway, I have empirical evidence that it work .... on my one Windows machine ;) Running collect strings resulted in no unstaged changes. Got all the expected output.

Reply:
> is node-glob === glob

[Yep](https://www.npmjs.com/package/glob#windows) :)

Just makes me nervous...

Reply:
apparently this is a whole thing: https://github.com/isaacs/minimatch/pull/109, https://github.com/isaacs/node-glob/issues/212 🤷‍♂

Reply:
aight, passing these to path.join normalizes to the OS's file sep.

Reply:
Which despite what the docs of node-glob state, it does seem to then convert to `/`

https://github.com/isaacs/node-glob/blob/e3fe170d9156e7c444a57b44a9e45e46c66f2dbf/glob.js#L701-L702

AFAICT :)